modules/features2d/misc/java/test/AKAZEDescriptorExtractorTest.java

```diff
@@ -0,0 +1,93 @@
+package org.opencv.test.features2d;
+
+import org.opencv.test.OpenCVTestCase;
+import org.opencv.test.OpenCVTestRunner;
+import org.opencv.features2d.AKAZE;
+
+public class AKAZEDescriptorExtractorTest extends OpenCVTestCase {
+
+    AKAZE extractor;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        extractor = AKAZE.create(); // default (5,0,3,0.001f,4,4,1)
+    }
+
+    public void testCreate() {
+        assertNotNull(extractor);
+    }
+
+    public void testDetectListOfMatListOfListOfKeyPoint() {
+        fail("Not yet implemented");
+    }
+
+    public void testDetectListOfMatListOfListOfKeyPointListOfMat() {
+        fail("Not yet implemented");
+    }
+
+    public void testDetectMatListOfKeyPoint() {
+        fail("Not yet implemented");
+    }
+
+    public void testDetectMatListOfKeyPointMat() {
+        fail("Not yet implemented");
+    }
+
+    public void testEmpty() {
+        fail("Not yet implemented");
+    }
+
+    public void testRead() {
+        String filename = OpenCVTestRunner.getTempFileName("xml");
+        writeFile(filename, "<?xml version=\"1.0\"?>\n<opencv_storage>\n<format>3</format>\n<name>Feature2D.AKAZE</name>\n<descriptor>4</descriptor>\n<descriptor_channels>2</descriptor_channels>\n<descriptor_size>32</descriptor_size>\n<threshold>0.002</threshold>\n<octaves>3</octaves>\n<sublevels>5</sublevels>\n<diffusivity>2</diffusivity>\n</opencv_storage>\n");
+
+        extractor.read(filename);
+
+        assertEquals(4, extractor.getDescriptorType());
+        assertEquals(2, extractor.getDescriptorChannels());
+        assertEquals(32, extractor.getDescriptorSize());
+        assertEquals((double)((float)0.002), extractor.getThreshold());
```

Comment:
Looks very suspicious. Please use assertEquals overload for floating point values with epsilon.